### PR TITLE
Create contact section

### PR DESCRIPTION
**Pull request by sancraz**

### CODE REVIEW

File: src/components/ContactButtons.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { FormButtonsContainer, AttachButton, ContactButton } from './ui';
```
Comment (DmitriyBodlev) [inline src/components/ContactButtons.js]: Cascade order imports `{ AttachButton, ContactButton, FormButtonsContainer }`

File: src/components/Contacts.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import React, { Fragment } from 'react';
+import { compose, withState, withHandlers } from 'recompose';
+
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js]: delete empty line

File: src/components/Contacts.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+// global
+import * as GC from '../constants';
+// components
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js:6]: components imports must be higher than global

File: src/components/Contacts.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    type: 'text',
+    name: 'details',
+    component: (props) => <FormTextArea {...props} />,
+  },
+];
+
+const initialFieldValues = {
+  yourName: '',
+  email: '',
+  phoneNumber: '',
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js:51]: cascade

File: src/components/Contacts.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  details: '',
+};
+
+const initialErrors = {
+  yourName: false,
+  email: false,
+  phoneNumber: false,
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js:58]: cascade

File: src/components/Contacts.js [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  details: false,
+};
+
+const getErrors = (values) => ({
+  yourName: !GC.USER_NAME_REGEX.test(values.yourName),
+  email: !GC.EMAIL_REGEX.test(values.email),
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js:63]: cascade

File: src/components/Contacts.js [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+});
+
+const enhance = compose(
+  withState('fields', 'setFields', { ...initialFieldValues }),
+  withState('errors', 'setErrors', { ...initialErrors }),
+  withState('requestPending', 'setRequestPending', false),
+  withHandlers({
+    handleHideError: (props) => (name) => {
+      const newErrors = props.errors;
+      newErrors[name] = false;
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js:77]: Do we use Ramda?

File: src/components/Contacts.js [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    },
+    handleFieldChange: (props) => (e, name) => {
+      const field = e.currentTarget;
+      const newFields = props.fields;
+      newFields[name] = field.value;
+      props.setFields(newFields);
+    },
+    handleSubmit: (props) => () => {
+      const errors = getErrors(props.fields);
+      const isInvalid = Object.values(errors).some(err => err === true);
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js]: isInvalid not correct name for boolean. use just `invalid`. Name with `is` must be a function

File: src/components/Contacts.js [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        contactButtonText={H.getLocaleItem(['send'], props.locale)}
+        attachButtonText={H.getLocaleItem(['attachFile'], props.locale)}
+      />
+    </FormContainer>
+    {
+      props.allowContactSection
+      && (
+      <ContactSection>
+        <ContactsInfo>
+          Email<br />bukbook.unity@gmail.com<br />
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js]: don't use <br />
make like
Talk with Sacha to use common styled component for this case instead div
    `<div>Email</div>`
    `<div>bukbook.unity@gmail.com</div>`
    `<div>some.email@gmail.com</div>`

File: src/components/Contacts.js [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+    <SectionTitle
+      titleText={H.getLocaleItem(['contactUs'], props.locale)}
+    />
+    <FormContainer>
+      {
+        fields.map((field) => (
+          <FormField
+            field={field}
+            locale={props.locale}
+            value={props.fields[field.name]}
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js:150]: R.prop(field.name, props.fields)

File: src/components/SectionTitle.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import React, { Fragment } from 'react';
+// ui
+// import { Text, Input, TextArea } from '../ui';
```
Comment (DmitriyBodlev) [inline src/components/SectionTitle.js]: add comment
// TODO: Delete comments if it not used

File: src/components/SectionTitle.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+// import styled from 'styled-components';
+import { Title, Line } from './ui';
+//  /////////////////////////////////////////////////////////////////////////////////////////////////
+
+export const SectionTitle = (props) => (
+  <Fragment>
+    <Title>{props.titleText}
```
Comment (DmitriyBodlev) [inline src/components/SectionTitle.js]: `<Title>`
        `{props.titleText}`
        `<Line />`
    `</Title>`

File: src/components/ui.js [hunk 13]
```diff
@@ -9,8 +9,13 @@ import { ifElse } from '../helpers';
 import {
   Box,
   Flex,
+  // Text,
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: delete if not used

File: src/components/ui.js [hunk 14]
```diff
@@ -9,8 +9,13 @@ import { ifElse } from '../helpers';
...
+  Input,
+  TextArea,
   createMinWithMediaQuery,
-  createMaxWithMediaQuery } from '../ui';
+  createMaxWithMediaQuery,
+  // Button
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: here too

File: src/components/ui.js [hunk 15]
```diff
@@ -59,6 +64,8 @@ export const HamburgerBtnWrapper = styled('div')`
   transform: rotate(0deg);
   transition: .5s ease-in-out;
 
+  
+
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: delete no needed empty spaces

File: src/components/ui.js [hunk 16]
```diff
@@ -114,6 +121,7 @@ export const BarNavigationWrapper = posed(styled(Box)`
   opened: {
     opacity: 1,
     y: '0px',
+    staggerChildren: 100,
```
Comment (DmitriyBodlev) [inline src/components/ui.js:119]: cascade

File: src/components/ui.js [hunk 17]
```diff
@@ -124,3 +132,149 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+
+export const FormInput = styled(Input)`
+  border: 0;
+  width: 262px;
+  font-size: 12px;
+  border-radius: 0;
+  margin-bottom: 15px;
+  font-family: "Roboto";
+  border-bottom: 1.5px solid #f18701; 
+  border-color: ${({ isInvalid }) => isInvalid && 'red'};
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: R.and(isInvalid, '#ff0000');

and change name on isInvalid in all places

File: src/components/ui.js [hunk 18]
```diff
@@ -124,3 +132,149 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+`;
+
+export const FormFieldContainer = styled.div`
+  position: relative;
```
Comment (DmitriyBodlev) [inline src/components/ui.js:195]: add empty line

File: src/components/ui.js [hunk 19]
```diff
@@ -124,3 +132,149 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  width: 262px;
+  resize: none;
+  height: 133.5px;
+  font-size: 12px;
+  font-family: "Roboto";
+  border-bottom: 1.5px solid #f18701;
+`;
+
+export const FormButtonsContainer = styled(Flex)`
+margin-top: 33px;
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: add right tabulation

File: src/components/ui.js [hunk 20]
```diff
@@ -124,3 +132,149 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  cursor: pointer;
+  font-size: 12px;
+  margin-right: 23px;
+  align-items: center;
+  font-family: "Roboto";  
+  justify-content: center;
+  background-color: #ffffff; 
+  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.24), 0 0 1px 0 rgba(0, 0, 0, 0.12);
+  
+    &::before {
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: mae right tabulation(- 2 spaces)

File: src/components/ui.js [hunk 21]
```diff
@@ -124,3 +132,149 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+`;
+
+export const ContactButton = styled(AttachButton)`
+  color: #ffffff;
+  font-size: 18px;
+  margin-right: 0;
+  margin-left: 23px;
+  background-color: #f18701;
+    
+    &::before {
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: here too tabulation

File: src/locale/index.js [hunk 22]
```diff
@@ -109,18 +140,30 @@ const RU = {
       to save and distribute the read... :)
     `,
   },
+  errorMessages: {
+    yourName: 'Please, enter a valid name RU',
+    email: 'Please, enter a valid email RU',
```
Comment (DmitriyBodlev) [inline src/locale/index.js:153]: cascade

File: src/locale/index.js [hunk 23]
```diff
@@ -72,18 +91,30 @@ const EN = {
...
     openBookbook: 'Open @bookbook',
   },
 };
 
 const RU = {
   home: 'home',
+  send: 'Send',
+  email: 'Email',
+  yourName: 'Name',
+  details: 'Details',
```
Comment (DmitriyBodlev) [inline src/locale/index.js]: why some names from Uppercase and others lowercase

File: src/locale/index.js [hunk 24]
```diff
@@ -146,6 +189,11 @@ const PL = {
       to save and distribute the read... :)
     `,
   },
+  errorMessages: {
+    yourName: 'Please, enter a valid name PL',
+    email: 'Please, enter a valid email PL',
```
Comment (DmitriyBodlev) [inline src/locale/index.js:205]: cascade

File: src/locale/index.js [hunk 25]
```diff
@@ -72,18 +91,30 @@ const EN = {
...
+  yourName: 'Name',
+  details: 'Details',
   aboutUs: 'about us',
   contacts: 'contacts',
-  termsAndConditions: 'terms and conditions',
+  contactUs: 'Contact Us',
+  attachFile: 'Attach File',
+  phoneNumber: 'Phone Number',
   privacyPolicy: 'privacy policy',
   copyright: '2018 © All rights reserved',
```
Comment (DmitriyBodlev) [inline src/locale/index.js]: 2019

File: src/components/ui.js [hunk 26]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+    top: 22px;
+    color: red;
+    font-size: 10px;
+    position: absolute;
+    white-space: nowrap;
+    content: '${({ text }) => text}';
+  }
+`;
+
+export const FormTextArea = styled(TextArea)`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js:208]: it's better to use reusable TextArea from global ui

File: src/components/ui.js [hunk 27]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  width: 262px;
+  resize: none;
+  height: 133.5px;
+  font-size: 12px;
+  font-family: "Roboto";
+  border-bottom: 1.5px solid #f18701;
+`;
+
+export const FormButtonsContainer = styled(Flex)`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: instead this component use Flex mt={33} mb={33} justifyContent='space-between'

File: src/components/ui.js [hunk 28]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  margin-top: 33px; 
+  margin-bottom: 33px; 
+  justify-content: space-between; 
+  `;
+
+export const AttachButton = styled(Flex)`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: Please, use reusable styled Button from global ui and use in all button cases (extend if you need)

File: src/components/ui.js [hunk 29]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  margin-right: 0;
+  margin-left: 23px;
+  background-color: #f18701;
+    
+    &::before {
+      content: "";
+    }
+`;
+
+export const ContactSection = styled(Flex)`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: use Flex directly

File: src/components/ui.js [hunk 30]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  z-index: 10;
+  display: flex;
+  position: fixed;
+  align-items: center;
+  justify-content: center;
+  background-color: rgba(0,0,0,0.3);
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: use from theme

File: src/components/ui.js [hunk 31]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+`;
+
+export const Loader = styled.div`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: use colors from theme and move to global ui

File: src/components/Contacts.js [hunk 32]
```diff
@@ -0,0 +1,200 @@
+import React, { Fragment } from 'react';
+import {
+  compose,
+  withState,
+  withHandlers,
+} from 'recompose';
```
Comment (VitaliyAndrushko) [inline src/components/Contacts.js]: move bracket to the same line as withHandlers. Check all import cases

File: src/components/Contacts.js [hunk 33]
```diff
@@ -0,0 +1,200 @@
...
+  FormContainer,
+  ContactSection,
+  FormFieldContainer,
+} from './ui';
+//  /////////////////////////////////////////////////////////////////////////////////////////////////
+
+const fields = [
+  {
+    type: 'text',
+    name: 'yourName',
```
Comment (VitaliyAndrushko) [inline src/components/Contacts.js]: rename to name

File: src/components/Contacts.js [hunk 34]
```diff
@@ -0,0 +1,200 @@
...
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'tel',
+    name: 'phoneNumber',
+    component: (props) => <FormInput {...props} />,
+  },
+  {
+    type: 'text',
+    name: 'details',
```
Comment (VitaliyAndrushko) [inline src/components/Contacts.js]: rename to message

File: src/components/ui.js [hunk 35]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+export const Line = styled.div`
+  bottom: 0; 
+  width: 47.5px; 
+  height: 4.5px; 
+  position: absolute; 
+  border-radius: 10px; 
+  background-color: #f18701;
+`;
+
+export const FormContainer = styled.div`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: use Flex instead of this component

File: src/components/ui.js [hunk 36]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  display: flex;
+  margin-top: 22px;
+  align-items: center;
+  flex-direction: column;  
+`;
+
+export const FormInput = styled(Input)`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js:184]: extend global Input with some props (like invalid) and use it

File: src/components/ui.js [hunk 37]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+    opacity: ${({ active }) => ifElse(active, 1, 0)};
+    color: ${themeGet('colors.mainOrange', 'white')}; 
+  }
+`)({
+  opened: { opacity: 1 },
+  closed: { opacity: 0 },
+});
+
+// Contacts form styles
+export const Title = styled(Flex)`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js:163]: use Box instead of this component

File: src/components/ui.js [hunk 38]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+    }
+`;
+
+export const ContactSection = styled(Flex)`
+  height: 205px;
+  position: relative;
+  background-color: #f1f1f1;
+`;
+
+export const ContactsInfo = styled.div`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: use Box and Text instead of it

File: src/components/ui.js [hunk 39]
```diff
@@ -124,3 +128,196 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  font-size: 15px;
+  margin-top: 19px;
+  line-height: 1.5;
+  margin-left: 31px;
+  position: absolute;
+  font-family: "Roboto"; 
+`;
+
+export const FadeContainer = styled.div`
```
Comment (VitaliyAndrushko) [inline src/components/ui.js]: move to global ui

File: src/components/Contacts.js (hunk 7 above)
Comment (sancraz) [reply src/components/Contacts.js:77]: we agree that we are trying to avoid using ramda

File: src/components/ContactButtons.js (hunk 1 above)
Comment (sancraz) [reply src/components/ContactButtons.js]: why? this is single line

File: src/components/Contacts.js (hunk 10 above)
Comment (sancraz) [reply src/components/Contacts.js:150]: As I mentioned above - not use ramda if not necessary. All fields are inside this component and error will mean we did something wrong

File: src/components/Contacts.js (hunk 8 above)
Comment (sancraz) [reply src/components/Contacts.js]: TODO: check everywhere isInvalid and change to invalid

File: src/locale/index.js (hunk 22 above)
Comment (sancraz) [reply src/locale/index.js:153]: let's not use cascade in locale to have the same each language props

File: src/components/ContactButtons.js (hunk 1 above)
Comment (p-pavlo) [reply src/components/ContactButtons.js]: +

File: src/components/Contacts.js (hunk 9 above)
Comment (p-pavlo) [reply src/components/Contacts.js]: TODO...

File: src/locale/index.js (hunk 22 above)
Comment (p-pavlo) [reply src/locale/index.js:153]: What I need to do? :)

File: src/components/Contacts.js (hunk 32 above)
Comment (p-pavlo) [reply src/components/Contacts.js]: I didn't get it. Please show an example.

Comment (p-pavlo) [reply src/components/Contacts.js]: TODO...

File: src/components/Contacts.js (hunk 33 above)
Comment (p-pavlo) [reply src/components/Contacts.js]: TODO...

File: src/components/Contacts.js (hunk 34 above)
Comment (p-pavlo) [reply src/components/Contacts.js]: TODO...

File: src/components/ui.js (hunk 35 above)
Comment (p-pavlo) [reply src/components/ui.js]: TODO...

File: src/components/ui.js (hunk 36 above)
Comment (p-pavlo) [reply src/components/ui.js:184]: TODO...

File: src/components/ui.js (hunk 37 above)
Comment (p-pavlo) [reply src/components/ui.js:163]: TODO...

File: src/components/ui.js (hunk 38 above)
Comment (p-pavlo) [reply src/components/ui.js]: TODO...

File: src/components/ui.js (hunk 39 above)
Comment (p-pavlo) [reply src/components/ui.js]: TODO...

File: src/components/ui.js (hunk 31 above)
Comment (p-pavlo) [reply src/components/ui.js]: TODO...

File: src/components/ui.js (hunk 30 above)
Comment (p-pavlo) [reply src/components/ui.js]: TODO...

File: src/components/ui.js (hunk 26 above)
Comment (p-pavlo) [reply src/components/ui.js:208]: TODO...

File: src/components/ContactButtons.js [hunk 40]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {
```
Comment (DmitriyBodlev) [inline src/components/ContactButtons.js:5]: import {
      Button,
      Flex } from '../ui';

File: src/components/ContactButtons.js [hunk 41]
```diff
@@ -0,0 +1,59 @@
...
+  Button,
+  Flex,
+} from '../ui';
+import {
```
Comment (DmitriyBodlev) [inline src/components/ContactButtons.js]: import { Paperclip } from './icons';

File: src/components/ContactButtons.js [hunk 42]
```diff
@@ -0,0 +1,59 @@
...
+  Paperclip,
+} from './icons';
+//  /////////////////////////////////////////////////////////////////////////////////////////////////
+
+export const ContactButtons = (props) => (
+  <Flex
+    mt={33}
+    mb={33}
+    width={262}
```
Comment (DmitriyBodlev) [inline src/components/ContactButtons.js]: why 262? use maybe 250 or 300

File: src/components/ContactButtons.js [hunk 43]
```diff
@@ -0,0 +1,59 @@
...
+          width={108}
+          height={35}
+          bg='#ffffff'
+          fontSize={12}
+          color='#f18701'
+          cursor='pointer'
+          alignItems='center'
+          fontFamily='Roboto'
+          justifyContent='center'
+          boxShadow='0 1px 1px 0 rgba(0, 0, 0, 0.24), 0 0 1px 0 rgba(0, 0, 0, 0.12)'
```
Comment (DmitriyBodlev) [inline src/components/ContactButtons.js]: add higher, before component comment  // TODO: With theme

File: src/components/ContactButtons.js [hunk 44]
```diff
@@ -0,0 +1,59 @@
...
+        >
+          <Paperclip
+            width='15px'
+            height='15px'
+          />
+          {props.attachButtonText}
+        </Button>
+        )
```
Comment (DmitriyBodlev) [inline src/components/ContactButtons.js:42]: tabulation

File: src/components/ContactButtons.js [hunk 45]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    <Button
+      onClick={props.onSubmit}
```
Comment (DmitriyBodlev) [inline src/components/ContactButtons.js]: props order

File: src/components/Contacts.js (hunk 32 above)
Comment (DmitriyBodlev) [reply src/components/Contacts.js]: import {
      compose,
      withState,
      withHandlers } from 'recompose';

File: src/components/Contacts.js [hunk 46]
```diff
@@ -0,0 +1,200 @@
...
+  Loader,
+  FormInput,
+  ResultWindow,
+  FormTextArea,
+  ContactsInfo,
+  FadeContainer,
+  FormContainer,
+  ContactSection,
+  FormFieldContainer,
+} from './ui';
```
Comment (DmitriyBodlev) [inline src/components/Contacts.js]: FormContainer,
    ContactSection,
    FormFieldContainer } from './ui';

File: src/components/SectionTitle.js [hunk 47]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+// TODO: Delete comments if it not used
+// ui
+import {
```
Comment (DmitriyBodlev) [inline src/components/SectionTitle.js]: import { Line, Title } from './ui';

File: src/components/icons/paperclip.js [hunk 48]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export const Paperclip = (props) => (
+  <svg
+    xmlns='http://www.w3.org/2000/svg'
+    width={props.width || ''}
```
Comment (DmitriyBodlev) [inline src/components/icons/paperclip.js]: width={props.width || '24'}
height={props.height || '24'}

File: src/components/ui.js [hunk 49]
```diff
@@ -9,8 +9,11 @@ import { ifElse } from '../helpers';
 import {
   Box,
   Flex,
+  Input,
+  TextArea,
   createMinWithMediaQuery,
-  createMaxWithMediaQuery } from '../ui';
+  createMaxWithMediaQuery,
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: createMinWithMediaQuery,
    createMaxWithMediaQuery } from '../ui';

File: src/components/ui.js [hunk 50]
```diff
@@ -124,3 +128,158 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  height: 40px;
+  border-radius: 50%;
+  border: 8px solid #f3f3f3;
+  border-top: 8px solid #f18701;
+  border-bottom: 8px solid #f18701;
+  animation: spin 2s linear infinite;
+  -webkit-animation: spin 2s linear infinite;
+  
+  @-webkit-keyframes spin {
+  0% { -webkit-transform: rotate(0deg); }
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: tabulation

File: src/components/ui.js [hunk 51]
```diff
@@ -124,3 +128,158 @@ export const BarNavigationWrapper = posed(styled(Box)`
...
+  100% { -webkit-transform: rotate(360deg); }
+  }
+
+  @keyframes spin {
+  0% { transform: rotate(0deg); }
```
Comment (DmitriyBodlev) [inline src/components/ui.js]: tabulation

File: src/ui/index.js [hunk 52]
```diff
@@ -124,13 +128,72 @@ export const Button = styled.button`
   ${height}
   ${border}
   ${zIndex}
+  ${opacity}
   ${fontSize}
   ${boxShadow}
   ${borderColor}
   ${borderRadius}
+  cursor: ${({ cursor }) => or(cursor, 'initial')};
+  &::before {
```
Comment (DmitriyBodlev) [inline src/ui/index.js:137]: move it to custom button component that inherited(`styled(Button)`) from this button

**Pull request closed without merging**